File: leetcode/937.reorder-data-in-log-files.cpp
// 途中でjavaの方が便利ってなった

/*
 * @lc app=leetcode id=937 lang=cpp
 *
 * [937] Reorder Data in Log Files
 */
#include <sort>
#include <string>
#include <vector>
using namespace std;

// @lc code=start
class Solution {
public:
  vector<string> reorderLogFiles(vector<string> &logs) {
    vector<string> ans, digitLogs;
    for (auto &log : logs) {
      if (isDigitLog(log)) {
        digitLogs.emplace_back(log);
      } else {
        ans.emplace_back(log);
      }
    }

    // sort(ans.begin(), ans.end(), )
  }

  int getContentBeginIndex(string &log) { return log.find(' '); }

  bool isDigitLog(string &log) { return false; }

  bool asc(string &a, string &b) {}
};
// @lc code=end

File: leetcode/java/937.reorder-data-in-log-files.java
/*
 * @lc app=leetcode id=937 lang=java
 *
 * [937] Reorder Data in Log Files
 */

import java.util.*;

// @lc code=start
class Solution {
	public String[] reorderLogFiles(String[] logs) {
		ArrayList<String> ans = new ArrayList<String>();
		ArrayList<String> digitLogs = new ArrayList<String>();
		for (String log : logs) {
			if (isDigitLog(log)) {
				digitLogs.add(log);
			} else {
				ans.add(log);
			}
		}

		// sort
		Collections.sort(ans, new Comparator<String>() {
			@Override
			public int compare(String s1, String s2) {
				String[] s1Split = s1.split(" ", 2);
				String[] s2Split = s2.split(" ", 2);
				int cmp = s1Split[1].compareTo(s2Split[1]);
				if (cmp != 0) {
					return cmp;
				}
				return s1Split[0].compareTo(s2Split[0]);
			}
		});

		for (String log : digitLogs) {
			ans.add(log);
		}

		return ans.toArray(new String[ans.size()]);
	}

	boolean isDigitLog(String log) {
		String[] splitted = log.split(" ", 2);
		return splitted[1].charAt(0) >= '0' && splitted[1].charAt(0) <= '9';
	}
}
// @lc code=end
